### PR TITLE
Issues #48 #49 #50 #51 #52

### LINKED ISSUE #48: Add new hiring applications through Driver service
## 🚀 Feature request
Implement a RESTful endpoint to add new hiring applications

### LINKED ISSUE #49: Get all hiring applications through Driver service
## 🚀 Feature request
Implement a RESTful endpoint get all hiring applications

### LINKED ISSUE #50: Approve a hiring application through Driver service
## 🚀 Feature request
Implement a RESTful endpoint approve a hiring application

### LINKED ISSUE #51: Searching appropriate drivers through Driver service
## 🚀 Feature request
Subscribe to "**drivers.find**" queue in order to start a job for searching an appropriate driver

### LINKED ISSUE #52: Connect to DB through Driver service
## 🚀 Feature request
Connect to a SQL Database

### PR DESCRIPTION
**Pull request by davi7725**
## Checklist
- [x] Implement a RESTful endpoint to add new hiring applications
- [x] Implement a RESTful endpoint get all hiring applications 
- [x] Implement a RESTful endpoint approve a hiring application
- [x] Subscribe to "**drivers.find**" queue in order to start a job for searching an appropriate driver
- [x] Connect to a SQL Database.

### Issue Description
I have created a new .net core web API.
I have created a DBSchema for the service, which lives in a "DBSCHEMA" file residing in the root.
I have added a POST endpoint to add new hiring applications.
I have added a GET endpoint to get all the currently unaccepted hiring applications.
I have added a PUT endpoint to approve a currently unaccepted hiring application.
Added connection to MySQL within the project.
Added connection to RabbitMQ within the project.
Added code to listen to messages in a "driver.find" queue which will then start the process of looking at appropriate drivers for the job.
When drivers are found they are published to a "driver.found" queue as JSON.

### How to test
I have created a docker image of the driver service.
Start by executing the following docker commands:

- `docker network create testnet`

- `docker run --rm --name mysql -d -v ${PWD}/db_data:/var/lib/mysql -p 3306:3306 --network=testnet -e MYSQL_ROOT_PASSWORD=soft2019Backend mysql`

 (This should be almost the same command you normally use, but if you are running your command make sure to add the --network=testnet and --name mysql, these are important). **Make sure to execute the DBSCHEMA file against the database to have the proper data.**

- `docker run --rm --name rabbitmq --network=testnet -p 15672:15672 -p 5672:5672 rabbitmq:management`

- `docker run --rm --name driverservice --network=testnet -p 5002:5002 davi7816/testdriverservice`

You can test the following endpoints:
- POST `https://localhost:5002/Application/` with the following body content:
```
{
    "Name": "Jane Doe",
    "DateOfBirth": "1997-01-01",
    "LicenseNumber": "010197-000",
    "WeekDays": [0,1,2,3,4,5,6]
 }
```
and a "Content-Type" header with value _application/json; charset=UTF-8_

- GET `https://localhost:5002/Application/`
- PUT `https://localhost:5002/Application/{driverID}`

After all of these are started and endpoints hit, you can test rest of the functionality by heading to the rabbitMQ web interface at [http://localhost:15672/](http://localhost:15672/). In there head over to queues where you should see a `driver.find` queue, click it.
When the page loads you should see a publish area where you can insert the following command:

- `{"from":"1996-06-23","to":"1996-06-25"}`
This will create a new queue (driver.found) with a result, which you can check by heading over to that queue (like you did to the current one) and pressing "Get message(s)"

#### Issue reference
close #48  
close #49 
close #50 
close #51 
close #52

### CODE REVIEW

File: DriverService/Models/Application.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            cmd.Parameters.Add(new MySqlParameter
+            {
+                ParameterName = "@name",
+                DbType = DbType.String,
+                Value = Name,
+            });
+            cmd.Parameters.Add(new MySqlParameter
+            {
+                ParameterName = "@dateofbirth",
+                DbType = DbType.DateTime,
```
Comment (elit0451) [inline DriverService/Models/Application.cs]: Shouldn't this be just `Date` type as in the DBSCHEMA it is also of type **Date**?

File: DriverService/RabbitMQ/EventBusRabbitMQ.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                properties.Persistent = true;
+                properties.DeliveryMode = 2;
+
+                channel.ConfirmSelect();
+                channel.BasicPublish(exchange: "", routingKey: _queueName, mandatory: true, basicProperties: properties, body: body);
+                channel.WaitForConfirmsOrDie();
+
+                channel.BasicAcks += (sender, eventArgs) =>
+                {
+                    Console.WriteLine("Sent RabbitMQ");
```
Comment (elit0451) [inline DriverService/RabbitMQ/EventBusRabbitMQ.cs]: Should we have more info on what we sent?

File: DriverService/Models/Application.cs (hunk 1 above)
Comment (davi7725) [reply DriverService/Models/Application.cs]: This converts from the Database type to the C# type, and within the C# Model class, we are using DateTime. That is the reason for this code to be like this.

File: DriverService/RabbitMQ/EventBusRabbitMQ.cs (hunk 2 above)
Comment (davi7725) [reply DriverService/RabbitMQ/EventBusRabbitMQ.cs]: We sure can do that, but this was more for debugging purposes. In the end, it should actually only be sent to our Logger service-

**Pull request merged**